### PR TITLE
AP_OSD_Screen: make BLH RPM fixed length

### PR DESCRIPTION
**Pull request by shellixyz**

### CODE REVIEW

**Comment by giacomo892** [comment]
Thanks! LGTM

File: libraries/AP_OSD/AP_OSD_Screen.cpp [hunk 1]
```diff
@@ -1623,7 +1623,9 @@ void AP_OSD_Screen::draw_blh_rpm(uint8_t x, uint8_t y)
     if (!AP::esc_telem().get_rpm(0, rpm)) {
         return;
     }
-    backend->write(x, y, false, "%3.1f%c%c", uint16_t(rpm) * 0.001f, SYM_KILO, SYM_RPM);
+    float krpm = rpm * 0.001f;
+    const char *format = krpm < 9.95 ? "%.2f%c%c" : (krpm < 99.95 ? "%.1f%c%c" : "%.0f%c%c");
```
Comment (andyp1per) [inline libraries/AP_OSD/AP_OSD_Screen.cpp:1627]: Why not 10 and 100?

Comment (shellixyz) [reply libraries/AP_OSD/AP_OSD_Screen.cpp:1627]: Because it wouldn't switch the format at the right moment. 9.95 %.1f displays 10.0

**Review by andyp1per — APPROVED** [review]
Ok :)

**Review by tridge — APPROVED** [review]

**Pull request merged**